src/entity/object/AreaEffectCloud.php

```diff
@@ -0,0 +1,462 @@
+	 */
+	public function setReapplicationDelay(int $delay) : void{
+		$this->reapplicationDelay = $delay;
+		$this->networkPropertiesDirty = true;
+	}
+
+	protected function entityBaseTick(int $tickDiff = 1) : bool{
+		$hasUpdate = parent::entityBaseTick($tickDiff);
+
+		$this->age += $tickDiff;
+		$radius = $this->radius + ($this->radiusChangePerTick * $tickDiff);
+		if($radius < 0.5){
+			$this->flagForDespawn();
+			return true;
+		}
+		$this->setRadius($radius);
+		if($this->age >= $this->waiting){
+			if($this->age > $this->duration){
+				$this->flagForDespawn();
+				return true;
+			}
+			foreach($this->victims as $entityId => $expiration){
+				if($this->age >= $expiration){
+					unset($this->victims[$entityId]);
+				}
+			}
+			$this->setWaiting($this->age + self::WAIT_TIME);
+
+			/** @var Living[] $entities */
+			$entities = [];
+			$radiusChange = 0.0;
+			$durationChange = 0;
+			foreach($this->getWorld()->getCollidingEntities($this->getBoundingBox(), $this) as $entity){
+				if(!$entity instanceof Living || isset($this->victims[$entity->getId()])){
+					continue;
+				}
+				$entityPosition = $entity->getPosition();
+				$xDiff = $entityPosition->getX() - $this->location->getX();
+				$zDiff = $entityPosition->getZ() - $this->location->getZ();
+				if(($xDiff ** 2 + $zDiff ** 2) > $this->radius ** 2){
+					continue;
+				}
+				$entities[] = $entity;
+				if($this->radiusChangeOnUse !== 0.0){
+					$radiusChange += $this->radiusChangeOnUse;
+					if($this->radius + $radiusChange <= 0){
+						break;
+					}
+				}
+				if($this->durationChangeOnUse !== 0){
+					$durationChange += $this->durationChangeOnUse;
+					if($this->duration + $durationChange <= 0){
+						break;
+					}
+				}
+			}
+			if(count($entities) === 0){
+				return $hasUpdate;
+			}
+
+			$ev = new AreaEffectCloudApplyEvent($this, $entities);
+			$ev->call();
+			if($ev->isCancelled()){
+				return $hasUpdate;
+			}
+
+			foreach($ev->getAffectedEntities() as $entity){
+				foreach($this->getCloudEffects() as $effect){
+					if($effect->getType() instanceof InstantEffect){
+						$effect->getType()->applyEffect($entity, $effect, 0.5, $this);
+					}else{
+						$entity->getEffects()->add($effect->setDuration((int) round($effect->getDuration() / 4)));
+					}
+				}
+				if($this->reapplicationDelay !== 0){
+					$this->victims[$entity->getId()] = $this->age + $this->reapplicationDelay;
+				}
+			}
+
+			$radius = $this->radius + $radiusChange;
+			$duration = $this->duration + $durationChange;
+			if($radius <= 0 || $duration <= 0){
+				$this->flagForDespawn();
+				return true;
+			}
+			$this->setRadius($radius);
+			$this->setDuration($duration);
+			$hasUpdate = true;
+		}
+
+		return $hasUpdate;
+	}
+
+	/**
+	 * Returns the effects the area effect cloud provides.
+	 *
+	 * Used to get COPIES to avoid accidentally modifying the same effect instance
+	 * already applied to another entity.
```

Comment:
No one is gonna read this. They'll just use `getEffects()` because that's what they're familiar with.